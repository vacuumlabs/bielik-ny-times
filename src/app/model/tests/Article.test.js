import {
  ARTICLE_FEATURED_IMAGE_WIDTH,
  ARTICLE_THUMBNAIL_SIZE,
  parseArticleResponse,
} from '../Article'

describe('Article', () => {
  describe('parseArticleResponse', () => {
    const makeArticleResponse = num => ({
      id: num,
      title: `title${num}`,
      abstract: `abstract${num}`,
      byline: `author${num}`,
      published_date: '2019-01-01',
      url: `url${num}`,
      media: [
        {
          type: 'image',
          'media-metadata': [
            {
              url: `url1`,
              width: ARTICLE_THUMBNAIL_SIZE,
              height: ARTICLE_THUMBNAIL_SIZE,
            },
            {
              url: `url2`,
              width: ARTICLE_FEATURED_IMAGE_WIDTH,
              height: ARTICLE_FEATURED_IMAGE_WIDTH,
            },
          ],
        },
      ],
    })
    it('should format response to custom object', () => {
      const response = {
        status: 'OK',
        copyright: 'copy',
        num_results: 2,
        results: [makeArticleResponse(1), makeArticleResponse(2)],
      }
      const actual = parseArticleResponse(response)
      expect(actual).toMatchSnapshot()
    })
  })
})
