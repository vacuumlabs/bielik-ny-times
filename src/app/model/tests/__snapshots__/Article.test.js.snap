// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Article parseArticleResponse should format response to custom object 1`] = `
Array [
  Object {
    "abstract": "abstract1",
    "authors": "author1",
    "featuredImage": Object {
      "height": 750,
      "url": "url2",
      "width": 750,
    },
    "id": 1,
    "published": "2019-01-01",
    "thumbnail": Object {
      "height": 60,
      "url": "url1",
      "width": 60,
    },
    "title": "title1",
    "url": "url1",
  },
  Object {
    "abstract": "abstract2",
    "authors": "author2",
    "featuredImage": Object {
      "height": 750,
      "url": "url2",
      "width": 750,
    },
    "id": 2,
    "published": "2019-01-01",
    "thumbnail": Object {
      "height": 60,
      "url": "url1",
      "width": 60,
    },
    "title": "title2",
    "url": "url2",
  },
]
`;
